### ISSUE TITLE
Extension not working with new board hub

### ISSUE THREAD

**Issue by itb-dev-bw** [ionpopsoi/azure-boards-decompose#1]
The extension seems not to be compatible with the new boards hub experience that is shortly rolling our for everyone.

1. Activate new boards hub experience from Preview Features menu
2. Open Decompose work item dialog, either from backlog context menu or from work item detail menu.
-> Tasks can be correctly entered
3. Click Create
-> Nothing happens. No Console log, not network request either.

Same with the old boards hub works good!

Can you fix it? Would be awesome!

**Comment by benhager** [comment]
We also see this issue, possible to address?

**Comment by sirforce** [comment]
I too observe the same issue where the create is no longer working.  There are several POST 400 errors that appear in the browser console to the Azure DevOps work item API.  Simply copying the URL to the browser and mimicking a GET request seems to return what looks like the expected data. I wonder if that is not indicative of the overall issue and perhaps an update to the API/library used to interact with devops simply needs a tweak or update.

I am going to pull the code and try to build an update and see if I can fix. If I can sort this out I am happy to share my findings and/or submit a PR with the caveat I've never worked on a DevOps extension before.

**Comment by Livieros** [comment]
Please bring the fix soon!

**Comment by danhellem** [comment]
@ionpopsoi are you able to look into this issue and fix it? Our Azure DevOps team can help you fix the extension if you need assistance.

**Comment by abstrqt** [comment]
sup folks!
currently using ver 1.1.10, Safari and cloud-based Azure Boards.
When i try to decompose the item something strange happens:
- extension sends POST request with empty Title
- gets 400
- no linger work
here is some clues from the Network tab



```
[{
    "op": "add",
    "path": "/fields/System.Title",
    "value": ""
}]

```

```
{
    "$id": "1",
    "customProperties": {
        "FieldReferenceName": null,
        "FieldStatusFlags": "none",
        "ErrorMessage": "TF401320: Rule Error for field Title. Error code: Required, InvalidEmpty.",
        "FieldStatusCode": 0,
        "RuleValidationErrors": [
            {
                "fieldReferenceName": "System.Title",
                "fieldStatusFlags": "required, invalidEmpty",
                "errorMessage": "TF401320: Rule Error for field Title. Error code: Required, InvalidEmpty.",
                "fieldStatusCode": 524289,
                "ruleValidationErrors": null
            }
        ]
    },
    "innerException": null,
    "message": "TF401320: Rule Error for field Title. Error code: Required, InvalidEmpty.",
    "typeName": "Microsoft.TeamFoundation.WorkItemTracking.Server.RuleValidationException, Microsoft.TeamFoundation.WorkItemTracking.Server",
    "typeKey": "RuleValidationException",
    "errorCode": 0,
    "eventId": 3200
}

```


```
Summary
URL: https://dev.azure.com/abstrqt/71f2c640-ee60-4877-86c5-5c974575b005/_apis/wit/workItems/$Feature?validateOnly=true
Status: 400
Source: Network
Address: 13.107.42.20:443
Initiator: 
Content:3:82719

```




<img width="1788" alt="Image" src="https://github.com/user-attachments/assets/f872de1e-90a5-4074-a7cd-649e87441ccc" />

after that the plugin stops to work

**Comment by itb-dev-bw** [comment]
> When i try to decompose the item something strange happens:
> [...]

Yeah, the extension is still just not compatible with new boards hub. That's all there is to it.

@ionpopsoi just forked and re-published it but never actually maintained it, and the original author @cschleiden abandoned it a long time ago. So now we are just waiting for someone to step up and take over... :\

**Comment by ryangurn** [comment]
I am also experiencing this issue. @abstrqt how did you manage to see the 400 response in your inspect element, I have been attempting to do that for a couple days now.

**Comment by ryangurn** [comment]
I have submitted [https://developercommunity.visualstudio.com/t/Better-workflow-for-creating-multiple-su/10876223](https://developercommunity.visualstudio.com/t/Better-workflow-for-creating-multiple-su/10876223) to microsoft, as I cannot believe that it is not just native functionality of the platform.

We are looking for up-votes over the next 90 days so there is a chance of convincing microsoft to add this to their roadmap sometime soon.

**Comment by teociaps** [comment]
Hey everyone, I saw the discussion here and wanted to jump in.

> So now we are just waiting for someone to step up and take over... :\

@itb-dev-bw To address this point— consider it done! 😄

I've been working on a revamped version of the extension that's **fully compatible with the new Boards Hub**, using the latest version of DevOps SDK and API.

It'll be a _free_ extension, and I'm planning to publish it in the next few weeks. 😉

Sometimes it feels like the community has to pick up the slack for features that should be native, right? 😅

I'll be sure to post the link here once it's live, and I'm committed to keeping this one maintained.
Just follow me if you want to be up to date and see when I'll make the extension (and its repo) public.

**Comment by teociaps** [comment]
Hey everyone!

Plot twist: after about a month (whoops, time flies!), I'm back!

For everyone following this thread, I've finally published my new extension: **Work Item Decomposer**

You can find it all here:
- Marketplace: [Work Item Decomposer](https://marketplace.visualstudio.com/items?itemName=teociaps.work-item-decomposer)
- GitHub: [teociaps/azdo-workitem-decomposer](https://github.com/teociaps/azdo-workitem-decomposer) (Repo & Wiki)

Hope this helps you out. Enjoy! 🚀


---

P.S. Tagging those who've been active on this issue to make sure you see this – sorry for the @-spam, but I figured you might have been waiting for this for a while! 😄 @benhager, @sirforce, @Livieros, @danhellem, @abstrqt, @itb-dev-bw, @ryangurn

**Comment by ryangurn** [comment]
@teociaps I'll see about giving it a test, thanks for your hard work

**Comment by ryangurn** [comment]
@teociaps, I am also always willing to assist if you need it. I am not the best at javascript development, but if you could ever share some time to knowledge dump resources, I am more than willing to assist in maintaining that package.

**Comment by itb-dev-bw** [comment]
Tested; works great, thanks @teociaps!

One of the main advantages of the old extension was that one could quickly add multiple tasks (or other sub-items) just by pressing Enter after each entry, without switching to the mouse. For example, type "Implement<Enter>Test<Enter>Document" and have three tasks.

Currently, one needs to switch to the mouse after each entry and hit the + button.
Could you re-implement this shortcut again? Enter = Add new work item on the same level as current focus.

**Comment by teociaps** [comment]
Thank you @itb-dev-bw!

Could you please open an issue, possibly as a feature request, on my repository? I think it would be better to close this thread.

In that way, I can handle it more effectively. Thanks!

(I can definitely do that tho! 😉)

Edit: [the previous comment related issue](https://github.com/teociaps/azdo-workitem-decomposer/issues/22)
